server/src/main/java/org/opensearch/common/settings/DoubleParser.java

```diff
@@ -0,0 +1,99 @@
+package org.opensearch.common.settings;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.function.Function;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.settings.Setting.Property;
+import org.opensearch.common.settings.Setting.Validator;
+
+public class DoubleParser<T> implements Function<String, T>, Writeable {
+    private Setting<Double> setting;
+    private Double minValue;
+    private Double maxValue;
+    private Double fallback;
+
+    public DoubleParser(Setting<Double> setting) {
+        this.setting = setting;
+    }
+
+    public DoubleParser(StreamInput in) throws IOException {
+        super();
+        // Read the key
+        String key = in.readString();
+        // Read the default value
+        Double defaultValue = in.readDouble();
+        // Read the min value
+        this.minValue = in.readDouble();
+        // Read the max value
+        this.maxValue = in.readDouble();
+        // Read the fallback
+        boolean hasFallback = in.readBoolean();
+        if (hasFallback) {
+            this.fallback = in.readDouble();
+        }
+        // Read properties
+        EnumSet<Property> propSet = in.readEnumSet(Property.class);
+        this.setting = doubleSetting(key, defaultValue, minValue, maxValue, propSet.toArray(Property[]::new));
+
+    }
+
+    public static Setting<Double> doubleSetting(
+        String key,
+        double defaultValue,
+        double minValue,
+        double maxValue,
+        Property[] properties
+    ) {
+        return new Setting<Double>(
+            key,
+            Double.toString(defaultValue),
+            (s) -> doubleParser(s, minValue, maxValue, key, isFiltered(properties)),
```

Comment:
this isn't really improving on the existing implementation, you're putting a non-`Writeable` object here for the parser.  The whole goal of what you are doing is to have `new Setting<Double>( ... writeableParser argument ... )`.  That way when you send `WriteableSetting` over transport you can just do `parser.writeTo(out)`.